### PR TITLE
BFD-2836: convert codegen annotation processor to plugin

### PR DESCRIPTION
**Pull request by brianburton**
**JIRA Ticket:**
[BFD-2836](https://jira.cms.gov/browse/BFD-2836)

**User Story or Bug Summary:**

As a BFD engineer I want to replace annotation processors with maven plugins to better integrate with IDEA.

---

### What Does This PR Do?

Replaces the codebook generation annotation processor and its related annotations with a single maven plugin that accomplishes the same task as part of the normal maven build process.  Refactors maven projects to give modules more appropriate names for their new role and eliminates obsolete code.

The files changed count is deceptive.  The number of significant changes is actually fairly small.  Most of the changes are just file renames or a trivial refactor like making a method non-static.  There were some fixes applied as well:

* fixing the use of `new Integer(n)`
* fixing identity equals check on a string value
* Converts previously static methods into instance methods so they can share configuration settings as fields on the instance.

Many other worthwhile improvements that could have been made were not as they were outside of the scope of this PR and would have made it hard to review.

Significant changes are:

* Removed the `bfd-model-codegen-annotations` module completely.
* Moves model classes from `bfd-model-codegen` into `bfd-model-codebook-library`.
* Replaces `bfd-model-codebook-extractor` module with `bfd-model-codebook-plugin`.
* Combines functionality from the `CodebookPdfToXmlApp` and `CodebookVariablesEnumProcessor` into `CodebookMojo` which uses `PdfParser` and a new `CodebookEnumGenerator` class.  The latter is just a small modification of logic from `CodebookVariablesEnumProcessor`.

Testing consisted of comparing the generated files from this branch to the ones generated from the master branch.  The only diff is the expected one in `CcwCodebookVariable.java`:

```
17572c17572
<   private static final Map<String, Variable> VARIABLES_BY_ID = new CodebookVariableReader().buildVariablesMappedById();
---
>   private static final Map<String, Variable> VARIABLES_BY_ID = CodebookVariableReader.buildVariablesMappedById();
```

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [x] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [x] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [x] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [x] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [x] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] The data dictionary has been updated with any field mapping changes, if any were made.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-model/bfd-model-codebook-plugin/src/test/java/gov/cms/bfd/model/codebook/extractor/PdfParserTest.java [hunk 1]
```diff
@@ -30,18 +36,118 @@
...
+        List<String> duplicatedCodes = new ArrayList<>();
+        for (String code : valuesByCode.keySet()) {
+          List<Value> values = valuesByCode.get(code);
+          if (values.size() > 1) duplicatedCodes.add(code);
+        }
+
+        // Log a detailed warning for each duplicate.
+        for (String duplicatedCode : duplicatedCodes) {
+          List<Value> duplicatedValues = valuesByCode.get(duplicatedCode);
+          LOGGER.warn(
```
Comment (lsmitchell) [inline apps/bfd-model/bfd-model-codebook-plugin/src/test/java/gov/cms/bfd/model/codebook/extractor/PdfParserTest.java:130]: This seems handy for debugging, but do we need this output on every test run? (I assume a warn is going to output with our default log levels)

**Review by lsmitchell — APPROVED** [review]
Overall pretty straightforward, and hopefully helps with intelliJ's compatibility with the project!

File: apps/bfd-model/bfd-model-codebook-plugin/src/test/java/gov/cms/bfd/model/codebook/extractor/PdfParserTest.java (hunk 1 above)
Comment (brianburton) [reply apps/bfd-model/bfd-model-codebook-plugin/src/test/java/gov/cms/bfd/model/codebook/extractor/PdfParserTest.java:130]: Good question.  That is a direct transfer from `SupportedCodebookTest`.  It logs all of that for me in master as well.  Maven hides the log output from tests so we never noticed it before.  I can see it in idea though.

File: apps/bfd-model/bfd-model-codebook-plugin/README.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
+# DSL Based Code Generation Maven Plugin
```
Comment (mackec1) [inline apps/bfd-model/bfd-model-codebook-plugin/README.md:1]: LOVE IT! so much cleaner/clearer with this.

**Review by mackec1 — APPROVED** [review]
While the '46 files changed' looked to be daunting, most are deletes or renames. As for the changes, well-thought out, and nice clean code. Much nicer to deal with what was a hack to begin with.

**Pull request merged**